.github/workflows/test.yml

```diff
@@ -0,0 +1,21 @@
+name: Test
+on:
+  pull_request:
+    branches: 
+      - master
+      - develop
+      - 'release/**'
+jobs: 
+  check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      
+      - name: install deps
+        run: yarn
+
+      - name: build
+        run: ./node_modules/lerna/cli.js run build
+        
+      - name: test
+        run: ./node_modules/lerna/cli.js run test
```

Comment:
Can you push a commit that will cause this command to fail? Am interested in seeing the failure output!

Reply:
Just pushed `4bc5510` which breaks stylelint

Reply:
Looks like it continued with the deploy - can we block the deploy on a successful `test` run?

Reply:
I think one valid use case for this is, "Can I see this with x and y changes?" during a design craft session for WIP.

While I'd love if everyone wrote code to style spec during their in-progress work, that seems like an unnecessary gate.

If we could hinge it on _certain_ tests failing, that seems more useful - e.g. don't deploy if you can't even build it, but deploy if the style rules are broken.

Reply:
We can, however, this functionality is by design. We want to encourage work-in-progress being shown to designers and other non-engineer folks. Often times, this may mean a test failure.

The reason we broke out the test workflow from the deployment workflow is to allow for this.

Reply:
I believe we see this feature used differently. I'd recommend **only** deploying docs to staging when the PR meets some set of criteria. At a _minimum_, linting is a good rule to enforce, followed closely by unit tests. As far as I can tell, our current test suite isn't that restrictive -- so this should mostly be a no-op.

As we build a more sophisticated test suite (unit, end-to-end, and integration tests) - then we can opt to have those run independently.

Reply:
For some clarity here, there is huge value for Design (and hopefully Engineering) to using this as a collaboration tool, not just a quality assurance or review system.

The intent as designed is to be able to workshop UI and implementation in the browser (both live and async) as opposed to the limits of Figma/Zoom today.

This will also act as a sign-off point for those involved, but that's only part of the value for us.

Reply:
Adding to that, I do think that this open-ended approach gives us the proverbial "win-win" in both the design _and_ engineering use-cases. An engineer can always pull down a branch and run it on their machine, this is expected of them. The same shouldn't be expected of a designer or other less "code-savvy" stakeholder